Add R113 blog post (#1068)
snowplow/snowplowanalytics.com: _posts/2019-02-18-snowplow-r113-filitosa-real-time-pipeline-improvements/index.md

- [Sven Pfenning][sven] and [Mirko Prescha][mirko] for the improvements made to our Kafka platform
- [Arun Manivannan](https://github.com/arunma), [Saeed Zareian](https://github.com/szareiangm) from
[the Globe and Mail](https://www.theglobeandmail.com/) and
[Toni Cebrián](https://github.com/tonicebrian) for the build improvements

Thanks a lot to everyone involved!

Please read on after the fold for:

1. [Scala Stream Collector improvements](#ssc)
2. [Scala Common Enrich improvements](#sce)
3. [Upgrading](#upgrading)
4. [Roadmap](#roadmap)
5. [Help](#help)

![filitosa][filitosa-img]
<br>
Jean-Pol Grandmont [CC BY-SA 3.0](https://creativecommons.org/licenses/by-sa/3.0)

<!--more-->

<h2 id="ssc">1. Scala Stream Collector improvements</h2>

<h3 id="metrics">1.1 Prometheus metrics support</h3>

Thanks to [LiveIntent][liveintent], the Scala Stream Collector now publishes
[Prometheus][prometheus] metrics to the `/metrics` endpoint. You'll find the following metrics
published at this endpoint:

- `http_requests_total`: the total count of requests
- `http_request_duration_seconds`: the time spent handling requests

> Which you'll be able to slice and dice by endpoint, method and/or response code.

Reviewer: ```suggestion
You will be able to slice and dice the metrics by endpoint, method and/or response code.
```

Additional information will also be available, such as the Java and Scala versions as well as the
version of the Scala Stream Collector artifact.

<h3 id="kafka">1.2 Improved Kafka support</h3>

It is now possible to specify arbitrary Kafka producer configurations for the collector through the
`collector.streams.sink.producerConf` configuration setting. Additionally, the Kafka library has
been upgraded to the latest version to leverage the latest features.

Note, that those changes are also true for Stream Enrich for Kafka through the
`enrich.streams.sourceSink.{producerConf, consumerConf}` configurations.

Thanks a lot to [Sven Pfenning][sven] and [Mirko Prescha][mirko] for those two awesome features!

<h3 id="ssc-other">1.3 Other improvements</h3>

For people using [the do not track cookie feature](https://snowplowanalytics.com/blog/2018/08/21/snowplow-r109-lambaesis-real-time-pipeline-upgrade/#dnt)
of the Scala Stream Collector, [LiveIntent][liveintent] has improved the feature by letting you

> specify a regex for the cookie value.

Reviewer: Can you think of a use case to add here?

[Mike][miike] from [Snowflake Analytics][sa] has introduced a configurable `Access-Control-Max-Age`
header which lets clients cache the results of `OPTIONS` request, resulting in fewer requests and
faster `POST` requests: no need to make a preflight request if the result is already cached.

<h2 id="sce">2. Scala Common Enrich improvements</h2>

<h3 id="hubspot">2.1 HubSpot webhook integration</h3>

[Peter Zhu][misterpig] from [Snowflake Analytics][sa] built the [HubSpot][hubspot] webhook

> integration from scratch for this release, huge props to him!

Reviewer: ```suggestion
integration from scratch for this release. Huge props to Peter!
```

You'll now be able to track the following HubSpot events in your Snowplow pipeline:

- Deal creation
- Deal change
- Deal deletion
- Contact creation
- Contact change
- Contact deletion
- Company creation
- Company change
- Company deletion

Improvements have also been made to the [Marketo](https://www.marketo.com/) and

> [CallRail](https://www.callrail.com/) integrations.

Reviewer: By Peter?

<h3 id="post">2.2 POST support in the API request enrichment</h3>

> It is now possible to use `POST` requests to interact with the API leveraged in the API request

Reviewer: Can you add what this is used for, Ben?
